tests/test_dte_data_models.py

```diff
@@ -222,70 +298,270 @@ def test_constants_match(self):
             DteDataL2.DATETIME_FIELDS_TZ,
         )
 
-    def test_init_fail(self) -> None:
-        # TODO: implement for 'DteDataL2()'
-        pass
+    def test_ok_razon_social_none(self) -> None:
+        _ = dataclasses.replace(
+            self.dte_l2_1,
+            emisor_razon_social=None,
+            receptor_razon_social=None,
+        )
 
-    def test_init_fail_razon_social_empty(self) -> None:
-        with self.assertRaises(ValueError) as cm:
+    def test_validate_emisor_razon_social_empty(self) -> None:
+        expected_validation_errors = [
+            {
+                'loc': ('emisor_razon_social',),
+                'msg': "Value must not be empty.",
+                'type': 'value_error',
+            },
+        ]
+
+        with self.assertRaises(pydantic.ValidationError) as assert_raises_cm:
             dataclasses.replace(
                 self.dte_l2_1,
                 emisor_razon_social='',
             )
-        self.assertEqual(cm.exception.args, ("Value must not be empty.", ))
-        with self.assertRaises(ValueError) as cm:
+
+        validation_errors = assert_raises_cm.exception.errors()
+        self.assertEqual(len(validation_errors), len(expected_validation_errors))
+        for expected_validation_error in expected_validation_errors:
+            self.assertIn(expected_validation_error, validation_errors)
+
+    def test_validate_receptor_razon_social_empty(self) -> None:
+        expected_validation_errors = [
+            {
+                'loc': ('receptor_razon_social',),
+                'msg': "Value must not be empty.",
+                'type': 'value_error',
+            },
+        ]
+
+        with self.assertRaises(pydantic.ValidationError) as assert_raises_cm:
             dataclasses.replace(
                 self.dte_l2_1,
                 receptor_razon_social='',
             )
-        self.assertEqual(cm.exception.args, ("Value must not be empty.", ))
 
-    def test_init_ok_razon_social_none(self) -> None:
-        _ = dataclasses.replace(
-            self.dte_l2_1,
-            emisor_razon_social=None,
-            receptor_razon_social=None,
-        )
+        validation_errors = assert_raises_cm.exception.errors()
+        self.assertEqual(len(validation_errors), len(expected_validation_errors))
+        for expected_validation_error in expected_validation_errors:
+            self.assertIn(expected_validation_error, validation_errors)
 
-    def test_init_fail_regression_signature_value_bytes_with_x20(self) -> None:
```

Comment:
Is this test method not needed anymore?

Reply:
replaced by `DteDataL2Test.test_validate_non_empty_bytes_signature_value`

Reply:
The test is still needed because it's a [regression test](https://en.wikipedia.org/wiki/Regression_testing) for a bug fix (ref: https://github.com/fyntex/lib-cl-sii-python/pull/40).

Reply:
You are absolutely right, my mistake was not trying to better understand the test before replacing it

Reply:
👍